lib/portage/tests/resolver/test_alternatives_gzip.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+from portage.tests import TestCase
+from portage.tests.resolver.ResolverPlayground import (
+    ResolverPlayground,
+    ResolverPlaygroundTestCase,
+)
+
+
+class AlternativesGzipTestCase(TestCase):
+    def testAlternativesGzip(self):
+        """
+        Test bug 917259, where app-alternatives/gzip was upgraded before
+        its pigz RDEPEND was installed.
+        """
+        ebuilds = {
```

Comment:
Yeah, I imagine we need some deep `@system` deps in-between in order to get the bad behaviour to reproduce.

If I had to guess, maybe it involved pigz depending on zlib, but dunno. Not going to worry about it unless we can reproduce it, as the problem is pretty well-understood.